src/app/api/chapter-request/route.ts

```diff
@@ -0,0 +1,85 @@
+import { NextRequest, NextResponse } from "next/server";
+import { ChapterRequest, ChapterRequestResponse } from "./route.schema";
+import { prisma } from "@server/db/client";
+
+export const POST = async (request: NextRequest) => {
+  // Validate the data in the request
+  // If the data is invalid, return a 400 response
+  // with a JSON body containing the validation errors
+
+  const chapterRequest = ChapterRequest.safeParse(
+    // Validate a proper JSON was passed in as well
+    await request.json().catch(() => {
+      return NextResponse.json(
+        ChapterRequestResponse.parse({
+          code: "UNKNOWN",
+          message: "Unknown error received",
+        }),
+        { status: 500 }
+      );
+    })
+  );
+  if (!chapterRequest.success) {
+    const error = chapterRequest.error.format();
+    // Check if any field fails the schema
+    if (Object.values(error).some((field) => !!field)) {
```

Comment:
Can you elaborate on this logic? If `ChapterRequest` fails to parse, wouldn't this line always evaluate to true?